### PR TITLE
feat(WAF): waf support rule cc protection resource

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
waf support rule cc protection resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleCCProtection_'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleCCProtection_ -timeout 360m -parallel 4 
=== RUN   TestAccRuleCCProtection_basic 
=== PAUSE TestAccRuleCCProtection_basic 
=== RUN   TestAccRuleCCProtection_withEpsID
=== PAUSE TestAccRuleCCProtection_withEpsID
=== CONT  TestAccRuleCCProtection_basic
=== CONT  TestAccRuleCCProtection_withEpsID
--- PASS: TestAccRuleCCProtection_withEpsID (398.65s) 
--- PASS: TestAccRuleCCProtection_basic (398.95s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       399.008s
```

### CODE REVIEW

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleCCProtection_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleCCProtection_basic -timeout 360m -parallel 4 
=== RUN   TestAccRuleCCProtection_basic 
=== PAUSE TestAccRuleCCProtection_basic
=== CONT  TestAccRuleCCProtection_basic
--- PASS: TestAccRuleCCProtection_basic (368.99s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       369.042s
```

File: docs/resources/waf_rule_cc_protection.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  The value ranges from 0 to 65,535 in seconds. This parameter is valid when `protective_action` is set to `block`.
+
+* `request_aggregation` - (Optional, Bool) Specifies whether to enable domain aggregation statistics.
+  This parameter is valid when `rate_limit_mode` is not set to `policy`. Default to false.
+
+* `all_waf_instances` - (Optional, Bool) Specifies whether to enable global counting. Default to false.
+
+* `description` - (Optional, String) Specifies the description of WAF cc protection rule.
+
+* `status` - (Optional, Int) Specifies the status of WAF cc protection rule.
```
Comment (hmbig2) [inline docs/resources/waf_rule_cc_protection.md:123]:
```suggestion
* `status` - (Optional, String) Specifies the status of WAF cc protection rule.
```
if the type is int, cant modify to 0

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleCCProtection_'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleCCProtection_ -timeout 360m -parallel 4 
=== RUN   TestAccRuleCCProtection_basic 
=== PAUSE TestAccRuleCCProtection_basic
=== RUN   TestAccRuleCCProtection_withEpsID
=== PAUSE TestAccRuleCCProtection_withEpsID
=== CONT  TestAccRuleCCProtection_basic
=== CONT  TestAccRuleCCProtection_withEpsID
--- PASS: TestAccRuleCCProtection_withEpsID (373.68s) 
--- PASS: TestAccRuleCCProtection_basic (375.24s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       375.293s
```

File: docs/resources/waf_rule_cc_protection.md (hunk 1 above)
Comment (deer-hang) [reply docs/resources/waf_rule_cc_protection.md:123]: type can change to 0 with current implementation.

**Pull request merged**